aips/aip-x.md

```diff
@@ -0,0 +1,707 @@
+    signature: deserializeSignInSignature(type, signature),
+    message,
+  });
+
+  if (!verification.valid) return c.json({ errors: verification.errors }, 400);
+
+  // ... create and store a session for the user
+
+  return c.json({ data: verification });
+});
+```
+
+4. Once the verification endpoint has been created and the wallet successfully responds to the application, verify the response with the new verification endpoint.
+
+```tsx
+export default function SignInPage() {
+  const { signIn } = useWallet();
+
+  const handleSignIn = async () => {
+    const input = await fetch("/auth/siwa");
+
+    const response: AptosSignInOutput = await signIn(input);
+
+    const verification = await fetch("/auth/siwa/verify", {
+      method: "POST",
+      body: JSON.stringify({
+        type: response.message.type,
+        signature: response.signature.bcsToHex().toString(),
+        message: response.message,
+        publicKey: response.publicKey.bcsToHex().toString(),
+      }),
+    });
+
+    if (!verification.ok) throw new Error("Failed to verify message");
+
+    // ... the user is signed in, redirect to application
+  };
+
+  // ...
+}
+```
+
+## Testing
+
+The new `@aptos-labs/siwa` package will have its utilities tested to ensure that validation is done correctly. Coverage is most important here. In addition, signature verification methods will be used from the `@aptos-labs/ts-sdk`, it's important that public key verification methods are supported and tested.
+
+## Risks and Drawbacks
+
+The ecosystem is responsible for adopting the new standard as an alternative to the `connect` + `signMessage` flow. If applications/wallets do not properly implement the `signIn` flow, they may suffer vulnerabilities that can lead to potential phishing, replay, or forgery attacks. If vulnerabilities are found within _Aptos Labs_ or ecosystem utilities, it’s important to patch and deprecate them as soon as possible. Applications that accept the `connect` + `signMessage` flow may still be vulnerable to security attacks.
+
+## Security Considerations
+
+It is important that signature verification methods are implemented correctly through the `@aptos-labs/ts-sdk`. If not correct, applications and wallets utilizing these methods will be vulnerable to attacks. Additional utilities provided from the `@aptos-labs/siwa` must be well tested to avoid attacks.
+
+The attack vectors that must be accounted for are the following:
+
+- Phishing attacks — A user must have visibility on signing a message that is for a different application.
+- Replay attacks — A hacker must not be able to use a signature outside of its intended usage.
+- Forgery attacks — A hacker must not be able to provide a false signature that can bypass verification.
+
+## Future Potential
+
+This AIP formally introduces a standard for authentication via Aptos decentralized identities. In the future, the standard may be extended to provide additional information from the wallet.
```

Comment:
> "via Aptos decentralized identities."

This is not precise. Unclear what "decentralized identity" means. It's a fluff umbrella term, AFAICT.

"via Aptos accounts" would be precise.